### ISSUE TITLE
`Giscus` support with `privacy` plugin

### ISSUE THREAD

**Issue by ZhiyuanChen** [squidfunk/mkdocs-material#3662]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

`Giscus` is not usable when enabling `privacy` plugin

### Expected behaviour

pass

### Actual behaviour

```
/zh-CN/widget?origin=https%3A%2F%2Fzhiyuanchen-project-zyc-ai-blog-rp79-8000.githubpreview.dev%2F%23activities&session=&theme=dark&reactionsEnabled=1&emitMetadata=1&inputPosition=top&repo=zyc-ai%2Fblog&repoId=MDEwOlJlcG9zaXRvcnk5OTQ5NTIwMg%3D%3D&category=General&categoryId=DIC_kwDOBe4tIs4CBXji&description=The+website+of+Zhiyuan+Chen&term=index
            HTTP/1.1" code 404
```

### Steps to reproduce

pass

### Package versions

- Python: `python 3.8.6`
- MkDocs: `mkdocs 1.2.3`
- Material: `8.2.3+insiders.4.10.0`

### Configuration

```yaml
pass
```

### System information

pass

**Comment by squidfunk** [comment]
Thanks for reporting. I think we should add an exclusion list for the plugin, so scripts and/or domains can be excluded. [Giscus self-hosting](https://github.com/giscus/giscus/blob/main/SELF-HOSTING.md) is challenging and needs more work than just inlining the script.

**Comment by squidfunk** [comment]
I've added exclusion patterns in [`36af41a`](https://github.com/squidfunk/mkdocs-material-insiders/commit/36af41a60fc7ecb79a8d62cda9ce2fed2579d020). You can now specify Unix-style file name patterns which are matched against all external scripts to determine whether a script should be excluded from being bundled, and thus left as-is. In order to exclude Giscus and MathJax from bundling, you can use the following configuration:

``` yaml
  - privacy:
      externals_exclude:
        - cdn.jsdelivr.net/npm/mathjax@3/*
        - giscus.app/*
```

I'll add documentation for this new feature tomorrow and issue a new release, but it's already usable on `master`. Also note that running MkDocs in `verbose` mode will report all excluded assets: `mkdocs build -v`

**Comment by squidfunk** [comment]
I've also updated the JSON schema in a2f84565c.

**Comment by squidfunk** [comment]
[Documentation](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#externals-exclude) was added in 1d9d6782d. I'll issue a new Insiders release later this day.

**Comment by squidfunk** [comment]
Released as part of 8.2.5+insiders-4.11.0.
